website/blog/2024-07-08-building-your-semantic-layer-in-pieces.md

```diff
@@ -0,0 +1,96 @@
+
+- It’s **duplicative, patchy, and confusing** as discussed above.
+- It’s **less powerful**.
+- You **can’t** **test** it.
+- Depending on the tool, oftentimes you **can’t** **version control** it.
+
+What you want is a unified development flow that handles **normalized transformation in dbt models** and **dynamic denormalization in the dbt Semantic Layer**.
+
+:::info
+🏎️ **The Semantic Layer is a denormalization engine.** dbt transforms your data into clean, normalized marts. The dbt Semantic Layer is a denormalization engine that dynamically connects and molds these building blocks into the maximum amount of shapes available _dynamically_.
+:::
+
+This enables a more **flexible consumption layer**, meaning downstream tools (like AI or dashboards) can sit as directly on top of Semantic Layer-generated artifacts and APIs as possible, and focus on what makes them shine instead of being burdened by basic dynamic modeling and aggregation tasks. Any tool-specific constructs should typically operate as close to **transparent pass-throughs** as you can make them, primarily serving to surface metrics and dimensions from the Semantic Layer in your downstream tool. There may be exceptions of course, but as a general guiding principle this gets you the most dynamic denormalization ability, and thus value, from your Semantic Layer code.
+
+So now we’ve established the system, let’s dig into the _plan_ for how we can get there iteratively.
+
+## The plan: towards iterative velocity
+
+1. **Identify a Data Product that is impactful**. Find something that is in heavy use and high value, but fairly narrow scope. **Don’t start with a broad executive dashboard** that shows metrics from across the company, because you’re looking to optimize for migrating the **smallest amount of modeling for the highest amount of impact** that you can. If there’s a dashboard focused on CAC for example, that relies on a narrow set of metrics and underlying tables that are nonetheless critical for your company, that would be a good starting place.
+2. **Catalog the models and their columns that service the Data Product**, both **in dbt _and_ the BI tool**, including rollups, metrics tables, and marts that support those. Pay special attention to aggregations as these will constitute _metrics_. You can reference [this example Google Sheet](https://docs.google.com/spreadsheets/d/1BR62C5jY6L5f5NvieMcA7OVldSFxu03Y07TG3waq0As/edit?usp=sharing) for one-way you might track this.
+3. [**Melt the frozen rollups**](https://docs.getdbt.com/best-practices/how-we-build-our-metrics/semantic-layer-6-terminology) in your dbt project, as well as variations modeled in your BI tool, **into Semantic Layer code.** We’ll go much more in-depth on this process, and we encourage you to read more about this tactical terminology (frozen, rollup, etc) in the link, it will be used throughout this article!
```

Comment:
```suggestion
3. [**Melt the frozen rollups**](https://docs.getdbt.com/best-practices/how-we-build-our-metrics/semantic-layer-6-terminology) in your dbt project, as well as variations modeled in your BI tool, **into Semantic Layer code.** We’ll go much more in-depth on this process, and we encourage you to read more about this tactical terminology (frozen, rollup, etc) in the link &mdash; it will be used throughout this article!
```